Content.Server/Disease/DiseaseSystem.cs

```diff
@@ -0,0 +1,420 @@
+                    RemComp<DiseasedComponent>(tuple.Item1.Owner);
+                tuple.Item1.PastDiseases.Add(tuple.Item2);
+                tuple.Item1.Diseases.Remove(tuple.Item2);
+            }
+            CureQueue.Clear();
+
+            foreach (var (diseasedComp, carrierComp, mobState) in EntityQuery<DiseasedComponent, DiseaseCarrierComponent, MobStateComponent>(false))
+            {
+                if (mobState.IsDead())
+                {
+                    if (_random.Prob(0.005f * frameTime)) //Mean time to remove is 200 seconds per disease
+                        CureDisease(carrierComp, _random.Pick(carrierComp.Diseases));
+                    continue;
+                }
+
+                foreach(var disease in carrierComp.Diseases)
+                {
+                    var args = new DiseaseEffectArgs(carrierComp.Owner, disease, EntityManager);
+                    disease.Accumulator += frameTime;
+                    if (disease.Accumulator >= 1f)
+                    {
+                        disease.Accumulator -= 1f;
+                        foreach (var cure in disease.Cures)
+                            if (cure.Cure(args))
+                                CureDisease(carrierComp, disease);
+                        foreach (var effect in disease.Effects)
+                            if (_random.Prob(effect.Probability))
+                                effect.Effect(args);
+                    }
+                }
+            }
+        }
+        ///
+        /// Event Handlers
+        ///
+
+        /// <summary>
+        /// Used when something is trying to cure ANY disease on the target,
+        /// not for special disease interactions. Randomly
+        /// tries to cure every disease on the target.
+        /// </summary>
+        private void OnTryCureDisease(EntityUid uid, DiseaseCarrierComponent component, CureDiseaseAttemptEvent args)
+        {
+            foreach (var disease in component.Diseases)
+            {
+                var cureProb = ((args.CureChance / component.Diseases.Count) - disease.CureResist);
+                if (cureProb < 0)
+                    return;
+                if (cureProb > 1)
+                {
+                    CureDisease(component, disease);
+                    return;
+                }
+                if (_random.Prob(cureProb))
+                {
+                    CureDisease(component, disease);
+                    return;
+                }
+            }
+        }
+        /// <summary>
+        /// Called when a component with disease protection
+        /// is equipped so it can be added to the person's
+        /// total disease resistance
+        /// </summary>
+        private void OnEquipped(EntityUid uid, DiseaseProtectionComponent component, GotEquippedEvent args)
+        {
+            if (TryComp<ClothingComponent>(uid, out var clothing))
+                if (clothing.SlotFlags != args.SlotFlags)
+                    return;
```

Comment:
```suggestion
                if (clothing.SlotFlags & args.SlotFlags != args.SlotFlags)
                    return;
```

this is a bitflag so just checking equality isnt enough, clothing might have multiple flags